Formula/b/build2.rb

```diff
@@ -0,0 +1,240 @@
+    # sqlite = Formula["sqlite"]
+
+    # Note that the below steps are equivalent to invoking build.sh with the
+    # --local --private --system libpkgconf,libsqlite3 options. The only
+    # difference are a few customizations to find Homebrew's libpkgconf and
+    # libsqlite3.
+    #
+    # For background on these steps, see:
+    #
+    # https://build2.org/build2-toolchain/doc/build2-toolchain-install.xhtml
+
+    # Suppress loading of default options files.
+    #
+    ENV["BUILD2_DEF_OPT"] = "0"
+
+    # Bootstrap, stage 1.
+    #
+    # Note that we disable all warnings since we cannot do anything more
+    # granular during bootstrap stage 1.
+    #
+    cd "build2" do
+      system "make",
+             "-j", ENV.make_jobs,
+             "-f", "./bootstrap.gmake",
+             "CXX=#{ENV.cxx}", "CXXFLAGS=-w"
+      system "build2/b-boot", "--version"
+    end
+
+    # Bootstrap, stage 2.
+    #
+    # Force using Homebrew's libpkgconf rather than building libpkg-config
+    # from source.
+    #
+    # Note that the build2 driver bootstrapped at stage 1 above doesn't read
+    # .pc files. Thus, during the bootstrap stage 2, it is assumed that the
+    # headers and libraries of the system dependencies can be found by the
+    # compiler and linker automatically. Failed that we have to manually
+    # specify their header/library paths, which is what we do here for
+    # libpkgconf.
+    #
+    cd "build2" do
+      system "build2/b-boot", "-v",
+             "-j", ENV.make_jobs,
+             "config.cxx=#{ENV.cxx}",
+             "config.bin.lib=static",
+             "config.build2.libpkgconf=true",
+             "config.cc.poptions=-I#{pkgconf.include}/pkgconf",
+             "config.cc.loptions=-L#{pkgconf.lib}",
+             "build2/exe{b}"
+      mv "build2/b", "build2/b-boot"
+      system "build2/b-boot", "--version"
+    end
+
+    # Local build.
+    #
+    # Note that we don't build hermetically to avoid baking Homebrew-specific
+    # paths into the final dylibs.
+    #
+    system "build2/build2/b-boot", "-V",
+           "config.cxx=#{ENV.cxx}",
+           "config.cc.coptions=-O3",
+           # "config.cc.loptions=-L#{pkgconf.lib} -L#{sqlite.lib}",
+           "config.cc.loptions=-L#{pkgconf.lib}",
+           "config.bin.lib=shared",
+           "config.bin.rpath=#{lib}/build2",
+           "config.install.root=#{prefix}",
+           "config.install.private=build2",
+           "config.import.libsqlite3=",
+           "config.build2.libpkgconf=true",
+           "config.config.persist='config.*'@unused=drop",
+           "configure"
+
+    system "build2/build2/b-boot", "-v",
+           "-j", ENV.make_jobs,
+           "install:", "build2/", "bpkg/", "bdep/"
+
+    ENV.prepend_path "PATH", bin
+
+    system "command", "-v", "b"
+    system "command", "-v", "bpkg"
+    system "command", "-v", "bdep"
+
+    system "b",    "--version"
+    system "bpkg", "--version"
+    system "bdep", "--version"
```

Comment:
This shouldn't be here

Reply:
These are just some sanity checks (to make sure we don't pick up an unrelated toolchain from a different location). But I will drop them.

Reply:
Resolved in [44e1301](https://github.com/Homebrew/homebrew-core/pull/141550/commits/44e130191b977fa3b007d8dae23578aa42358695).